### PR TITLE
Draft version to support workflow

### PR DESCRIPTION
**Pull request by headcrabmeat**
1. Broken backward compatibility to send In-progress status in freestyle project
2. For testing purpose removed some logic in calling TokenMacro.expandAll(...)
3. Working fine in jenkins2.0 + pipleine
4. Tests are not yet updated.

### CODE REVIEW

**Comment by ssbarnea** [comment]
@headcrabmeat Any updates on this? I am quite interested in reviewing and testing this fix. At this moment it seems that it is not passing CI.

**Comment by headcrabmeat** [comment]
@ssbarnea, 

CI is broken as some tests are not passed but `*.jpi` file could be compiled as well. 
As I wrote above I'd broken some minor features in free-style project kind. 
However you can use this version of the plugin in your Jenkins setup it works well for both pipeline and freestyle.

**Comment by ariellevy** [comment]
any news about this pull request?

**Comment by PierreBtz** [comment]
@headcrabmeat 

I compiled and installed the version with your fix. I can confirm it's working with my freestyle jobs (I did not notice any issue...yet :)). 
I also did a test on a pipeline job using the sample you wrote in #92 and it also worked upon modifying the call to match the name of the enum:

```
public enum StashBuildState {
	SUCCESSFUL, 
	FAILED,
	INPROGRESS,
}
```
I just noticed the plugin does not appear in the pipeline syntax helper. Anyway thanks for your work!

File: src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java [hunk 1]
```diff
@@ -604,6 +626,7 @@ public boolean configure(
             prependParentProjectKey = formData.getBoolean("prependParentProjectKey");
 
 			disableInprogressNotification = formData.getBoolean("disableInprogressNotification");
+			buildState = formData.getString("buildState");
```
Comment (PierreBtz) [inline src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java]: As mentioned in #108, this causes a crash in Jenkins global configuration page. I'm not familiar with Jenkins plugin development, but I'm not sure you have to query the buildState here as this method seems to be used only to keep global configurations.

Comment (scaytrase) [inline src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java]: Also I think this is a bad architectural design to pass the build state to the notifier constructor. That's not a part of configuration, but a part of runtime call

**Comment by scaytrase** [comment]
Implemented with #115

**Pull request closed without merging**